aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -375,3 +461,61 @@ impl<'r, 'l, S> DerefMut for SessionExt<'r, 'l, S> {
         &mut self.inner
     }
 }
+
+struct WriteOpConverter<'r, 'l, S> {
+    remote: &'r MoveResolverWithVMMetadata<'r, 'l, S>,
+    new_slot_metadata: Option<StateValueMetadata>,
+}
+
+impl<'r, 'l, S: MoveResolverExt> WriteOpConverter<'r, 'l, S> {
+    fn convert(
+        &self,
+        state_key: &StateKey,
+        move_storage_op: MoveStorageOp<Vec<u8>>,
+        legacy_creation_as_modification: bool,
+    ) -> Result<WriteOp, VMStatus> {
+        use MoveStorageOp::*;
+        use WriteOp::*;
+
+        let existing_value_opt = self
+            .remote
+            .get_state_value(state_key)
```

Comment:
I need to access the metadata on the existing state value here. @zekun000  @gelash